src/dbnode/encoding/m3tsz/roundtrip_test.go

```diff
@@ -112,77 +106,100 @@ func validateRoundTrip(t *testing.T, input []ts.Datapoint, intOpt bool) {
 	defer ctx.Close()
 
 	encoder := NewEncoder(testStartTime, nil, intOpt, nil)
-	for j, v := range input {
-		if j == 0 {
-			encoder.Encode(v, xtime.Millisecond, proto.EncodeVarint(10))
-		} else if j == 10 {
-			encoder.Encode(v, xtime.Microsecond, proto.EncodeVarint(60))
-		} else {
-			encoder.Encode(v, xtime.Second, nil)
+	timeUnits := make([]xtime.Unit, 0, len(input))
+	annotations := make([]ts.Annotation, 0, len(input))
+
+	for i, v := range input {
+		timeUnit := xtime.Second
+		if i == 0 {
+			timeUnit = xtime.Millisecond
+		} else if i == 10 {
+			timeUnit = xtime.Microsecond
+		}
+
+		var annotation, annotationCopy ts.Annotation
+		if i < 5 {
+			annotation = ts.Annotation("foo")
+		} else if i == 10 {
+			annotation = ts.Annotation("bar")
+		}
```

Comment:
nit: can you add `if i < 7 { annotation = ts.Annotation("baz" }` to flex the case where an annotation changes on the next incoming point too?